Added Basic Matter Sample (#1)
krzysztofziobro/sdk-nrf: samples/matter/template/README.rst

.. _matter_template_sample:

Matter: Template
#################

.. contents::
   :local:
   :depth: 2

This sample demonstrates a minimal implementation of the :ref:`Matter <ug_matter>` (formerly Project Connected Home over IP, Project CHIP) application layer.
This basic implementation enables a device to be commissioned into a Matter network.
The main goal of this sample is to provide a basis for developing your own application.

Requirements
************

The sample supports the following development kits:

.. table-from-rows:: /includes/sample_board_rows.txt
   :header: heading
   :rows: nrf52840dk_nrf52840, nrf5340dk_nrf5340_cpuapp

> For remote testing scenarios, if you want to commission the device and :ref:`control it remotely <matter_template_network_mode>` through a Thread network, you also need the following:

Reviewer: ```suggestion
To test this sample you need to commission the device and :ref:`control it remotely <matter_template_network_mode>` through a Thread network using a Matter controller device :ref:`configured on PC or smartphone <ug_matter_configuring>`.
```
...since we don't describe any other testing scenario.

* A Matter controller device :ref:`configured on PC or smartphone <ug_matter_configuring>`.

.. note::
    |matter_gn_required_note|

Overview
********

The sample automatically starts the Bluetooth LE advertising and gets the Matter device ready for commissioning into a Thread network.

> It uses LED to show the state of the connection, and a button to allow factory reset when needed.

Reviewer: ```suggestion
It uses an LED to show the state of the connection, and a button to initiate the factory reset of the device.
```

> Testing requires a Matter controller, which can be configured either on PC or mobile (for remote testing in a network).

Reviewer: ```suggestion
Testing requires a Matter controller, which can be configured either on PC or mobile.
```
